app/controllers/concerns/course/forum/topic_controller_hiding_concern.rb

```diff
@@ -0,0 +1,31 @@
+module Course::Forum::TopicControllerHidingConcern
+  extend ActiveSupport::Concern
+
+  def set_hidden
+    authorize!(:set_hidden, @topic)
```

Comment:
can you check if this is necessary? i think it is not. check using a standard user.